src/Netscript/NetscriptHelpers.ts

```diff
@@ -248,20 +248,20 @@ function makeRuntimeErrorMsg(ctx: NetscriptContext, msg: string, type = "RUNTIME
   const stack = errstack.split("\n").slice(1);
   const ws = ctx.workerScript;
   const caller = ctx.functionPath;
-  const scripts = ws.getServer().scripts;
   const userstack = [];
   for (const stackline of stack) {
-    let filename;
-    for (const script of scripts) {
-      if (script.filename && stackline.includes(script.filename)) {
-        filename = script.filename;
-      }
-      for (const dependency of script.dependencies) {
-        if (stackline.includes(dependency.filename)) {
-          filename = dependency.filename;
-        }
+    function getFileName() {
+      // Filename is current file if url found
+      if (ws.scriptRef.url && stackline.includes(ws.scriptRef.url)) return ws.scriptRef.filename;
+      // Also check urls for dependencies
+      for (const [url, script] of ws.scriptRef.dependencies) if (stackline.includes(url)) return script.filename;
+      // Check for filenames directly if no URL found
+      if (stackline.includes(ws.scriptRef.filename)) return ws.scriptRef.filename;
+      for (const script of ws.scriptRef.dependencies.values()) {
+        if (stackline.includes(script.filename)) return script.filename;
       }
     }
+    const filename = getFileName();
```

Comment:
Probably clearer as an IIFE (Immediately Invoked Function Expression), that way you don't wonder why it's a named function.

Reply:
Done c1183c8